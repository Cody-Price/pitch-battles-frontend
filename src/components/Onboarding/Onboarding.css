.onboarding {
  font-family: "chronotype";
  background: white;
  width: 600px;
  height: 360px;
  background: url("./images/onboarding-frame.png");
  background-size: cover;
  image-rendering: pixelated;
  position: relative;
  margin: 3rem auto;
  z-index: 1000;
  display: flex;
  overflow: hidden;
  font-size: 1.3rem;
  animation: slide-onboarding 0.3s linear;
  -webkit-animation: slide-onboarding 0.3s linear;
  -moz-animation: slide-onboarding 0.3s linear;
  -o-animation: slide-onboarding 0.3s linear;
}

.article-wrapper {
  position: relative;
}

.article-row {
  display: flex;
  position: absolute;
  left: 0;
  transition: 0.4s;
}

.skip-to-dash {
  position: absolute;
  right: 32px;
  top: 0px;
  z-index: 3000;
}

.skip-to-dash:hover {
  cursor: pointer;
}

.arrow-left,
.arrow-right {
  position: absolute;
  top: 130px;
  z-index: 2000;

  height: 21px;
  width: 15px;
}

.arrow-left:hover,
.arrow-right:hover {
  cursor: pointer;
}

.arrow-left:focus,
.arrow-right:focus {
  outline: none;
}

.arrow-left {
  background: url(./images/arrow-left.png);
  left: 50px;
  background-size: cover;
  image-rendering: pixelated;
}

.arrow-right {
  background: url(./images/arrow-right.png);
  right: 50px;
  background-size: cover;
  image-rendering: pixelated;
}

.arrow-left.false,
.arrow-right.false {
  display: none;
}

.dots-wrapper {
  position: absolute;
  bottom: 20px;
  height: 24px;
  width: 104px;
  margin-left: auto;
  margin-right: auto;
  left: 0;
  right: 0;
  overflow: hidden;
}

.dots {
  position: absolute;
  background: url("./images/dot-row.png");
  background-size: cover;
  image-rendering: pixelated;
  width: 190px;
  height: 18px;
  right: 0px;
  transition: 0.4s;
}

.dots.two {
  right: -18px;
}

.dots.three {
  right: -36px;
}

.dots.four {
  right: -54px;
}

.dots.five {
  right: -70px;
}

.dots.six {
  right: -86px;
}

.article-row.one {
  left: 0;
}

.article-row.two {
  left: -600px;
}

.article-row.three {
  left: -1200px;
}

.article-row.four {
  left: -1800px;
}

.article-row.five {
  left: -2400px;
}

.article-row.six {
  left: -3000px;
}

@keyframes slide-onboarding {
  0% {
    right: 4000px;
  }
  100% {
    right: 0;
  }
}

@-webkit-keyframes slide-onboarding {
  0% {
    right: 4000px;
  }
  100% {
    right: 0;
  }
}

@-moz-keyframes slide-onboarding {
  0% {
    right: 4000px;
  }
  100% {
    right: 0;
  }
}

@-o-keyframes slide-onboarding {
  0% {
    right: 4000px;
  }
  100% {
    right: 0;
  }
}

@media (max-width: 700px) {
  .onboarding {
    width: 340px;
    height: 431px;
    background: url("../Landing/landingframe3.png");
    background-size: cover;
    image-rendering: pixelated;
    margin-top: 0;
    top: 2rem;
  }

  .article-row.one {
    left: 0;
  }

  .article-row.two {
    left: -340px;
  }

  .article-row.three {
    left: -680px;
  }

  .article-row.four {
    left: -1020px;
  }

  .article-row.five {
    left: -1360px;
  }

  .article-row.six {
    left: -1700px;
  }

  .dots-wrapper {
    bottom: 40px;
  }
}
