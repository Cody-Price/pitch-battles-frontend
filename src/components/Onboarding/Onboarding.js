import React, { Component } from "react";
import "./Onboarding.css";
import { onboardingContent } from "../../utilities/onboardingContent";
import { dotsHelper } from "../../utilities/dotsHelper";

import OnboardingArticle from "../OnboardingArticle/OnboardingArticle";

class Onboarding extends Component {
  constructor() {
    super();

    this.state = {
      dot: 0,
      left: false,
      right: true
    };
  }

  clickRight = () => {
    if (this.state.dot < 5) {
      this.setState({
        dot: this.state.dot + 1
      });
    }
  };

  clickLeft = () => {
    if (this.state.dot !== 0) {
      this.setState({
        dot: this.state.dot - 1
      });
    }
  };

  render() {
    const articles = onboardingContent.map(article => {
      return <OnboardingArticle data={article} key={article.key} />;
    });
    return (
      <main className="onboarding">
        <h3
          onClick={() => {
            this.props.navigate("student dash");
          }}
          className="skip-to-dash"
        >
          skip
        </h3>
        <div
          onClick={this.clickLeft}
          className={`arrow-left ${this.state.left}`}
        />
        <div
          onClick={this.clickRight}
          className={`arrow-right ${this.state.right}`}
        />
        <section className="article-wrapper">
          <section className={`article-row ${dotsHelper[this.state.dot]}`}>
            {articles}
          </section>
        </section>
        <div className="dots-wrapper">
          <div className={`dots ${dotsHelper[this.state.dot]}`} />
        </div>
      </main>
    );
  }
}

export default Onboarding;
