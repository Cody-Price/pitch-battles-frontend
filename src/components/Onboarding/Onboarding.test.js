import React from "react";
import ReactDOM from "react-dom";
import Onboarding from "./Onboarding";

it("is a test", () => {});
