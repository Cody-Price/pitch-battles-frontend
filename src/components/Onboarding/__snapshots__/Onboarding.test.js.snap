// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding should match the snapshot 1`] = `
<main
  className="onboarding"
>
  <h3
    className="skip-to-dash"
    onClick={[Function]}
  >
    skip
  </h3>
  <div
    className="arrow-left false"
    onClick={[Function]}
  />
  <div
    className="arrow-right true"
    onClick={[Function]}
  />
  <section
    className="article-wrapper"
  >
    <section
      className="article-row one"
    >
      <OnboardingArticle
        data={
          Object {
            "content": "In Pitch Battles, musicians can test their pitch-naming mettle in combat against beasts of the wilds.",
            "imageUrl": "./images/battle-ob.png",
            "key": "intro",
          }
        }
        key="intro"
      />
      <OnboardingArticle
        data={
          Object {
            "content": "Before charing in, students are encouraged to join a class on the student dashboard by clicking on the 'join a class' link and entering the key given to them by their teacher.",
            "imageUrl": "./images/join-class-ob.png",
            "key": "join-class",
          }
        }
        key="join-class"
      />
      <OnboardingArticle
        data={
          Object {
            "content": "To play pitch battle, choose your instrument from the menu and click 'to battle!'.",
            "imageUrl": "./images/db-ob.png",
            "key": "start-game",
          }
        }
        key="start-game"
      />
      <OnboardingArticle
        data={
          Object {
            "content": "Once the game begins, you will be shown a staff with a single note. Identify it corrently by either clicking the pitch name or by entering it on the keyboard an you will strike the beast!",
            "imageUrl": "./images/pitch-ob.png",
            "key": "play-the-game",
          }
        }
        key="play-the-game"
      />
      <OnboardingArticle
        data={
          Object {
            "content": "Guess incorrectly and the beast will strike you. Each hit results in a lost heart. Knock out all the beast's hearts to advance to the next level. Lose all your hearts and it's game over...",
            "imageUrl": "./images/main-game-ob.png",
            "key": "game-description",
          }
        }
        key="game-description"
      />
      <OnboardingArticle
        data={
          Object {
            "content": "Playing games, beating levels, and improving your times will earn you badges. View your badges on your dashboard or see your entire class's progress on the class page (accessed from the dashboard)",
            "imageUrl": "./images/db-ob.png",
            "key": "badges-description",
          }
        }
        key="badges-description"
      />
    </section>
  </section>
  <div
    className="dots-wrapper"
  >
    <div
      className="dots one"
    />
  </div>
</main>
`;
